### ISSUE TITLE
[CodeGenerator] Null component generation

### ISSUE THREAD

**Issue by zehreken** [sschmid/Entitas-CSharp#469]
I have a component named BattleComponent in a pool named Battle. I've decided to remove the component, and as I always do I removed the component, its generated extension and deleted the related line in BattleComponentIds. When I regenerate, Entitas adds null fields and also the totalComponents is wrong. If I create another component and remove it, there is no problem. Is there a cache somewhere that causes this issue?

```csharp
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by Entitas.CodeGenerator.ComponentIndicesGenerator.
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------
public static class BattleComponentIds {

    public const int BattleFinished = 0;
    public const int GS_Battle = 1;
    public const int Hero = 2;
    public const int Village = 3;
    public const int Timer = 5;

    public const int TotalComponents = 6;

    public static readonly string[] componentNames = {
        "BattleFinished",
        "GS_Battle",
        "Hero",
        "Village",
        null,
        "Timer"
    };

    public static readonly System.Type[] componentTypes = {
        typeof(BattleFinishedComponent),
        typeof(GS_BattleComponent),
        typeof(HeroComponent),
        typeof(VillageComponent),
        null,
        typeof(TimerComponent)
    };
}
```

**Comment by sschmid** [comment]
Hm, never heard that before. There's no caching going on, the code generator always inspects the dll and generates.

**Comment by zehreken** [comment]
I'm using 0.35.0 by the way. Yes it is really weird.

**Comment by sschmid** [comment]
oh, that might explain a lot :) I improved a lot in the last few releases. 0.35.0 comes with sources instead of dll, so you might be able to debug a little but. I never saw sth like that before and I cannot recall lot of the details of the previous generation of the code generator.

I wrote down a note for my Video Tutorials :) Entitas Shmup is 0.35.0 too. Maybe I should to a "how to upgrade from 0.35.0" Video.
